### PR TITLE
[Go] Fix overloaded functions with noncapitalized class as parameter type

### PR DESCRIPTION
**Pull request by dontpanic92**
#676

### CODE REVIEW

**Comment by ojwb** [comment]
@ianlancetaylor Could you review this when you get a chance?  It looks like a plausible fix to me, but I'm not very familiar with Go or the SWIG Go backend.

File: Source/Modules/go.cxx [hunk 1]
```diff
@@ -6525,8 +6525,10 @@ class GO:public Language {
 	break;
       }
     }
-
-    if (Getattr(undefined_types, ty) && !Getattr(defined_types, ty)) {
+    
+    String* go_type = goType(n, ty);
+    
+    if (Getattr(undefined_types, go_type) && !Getattr(defined_types, go_type)) {
```
Comment (ianlancetaylor) [inline Source/Modules/go.cxx]: As far as I can see, the index into `undefined_types` should be `ty` and the index into `defined_types` should be `go_type`.

Also you should `Delete(go_type)` somewhere.

**Comment by ianlancetaylor** [comment]
@ojwb Thanks for the pointer, I missed this.

**Comment by ojwb** [comment]
Sorry, it seems your final update got missed.

I've added a CHANGES,current entry and a testcase based on the reproducer in #676 and merged this.

**Pull request closed without merging**